refract-spec.md

```diff
@@ -12,89 +12,279 @@ This document conforms to RFC 2119, which says:
 
 [MSON](https://github.com/apiaryio/mson) is used throughout this document to describe and define data structures.
 
-## Data Structure
+## Primitive Types
 
-Refract provides a single structure for data, which will be referred to throughout this document as an element. All elements found in a Refract document SHOULD conform to the element structure as defined here.
+Primitive Types are types used by Refract as the basis for all values and structure. These types use both JSON primitive values along with Refract Elements.
```

Comment:
These types are the basis for Refract, much like a type is the basis for a programming language or structure like JSON. Refract types can be defined as JSON primitives or Refract Elements, which gives these types the ability to be expanded out into infinity.